youtube_dl/extractor/tiktok.py

```diff
@@ -0,0 +1,78 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..utils import (
+    compat_str,
+    int_or_none,
+    str_or_none,
+    try_get,
+    url_or_none,
+)
+
+
+class TikTokIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:m\.)?tiktok\.com/v/(?P<id>[0-9]+)'
+    _TEST = {
+        'url': 'https://m.tiktok.com/v/6606727368545406213.html',
+        'info_dict': {
+            'id': '6606727368545406213',
+            'ext': 'mp4',
+            'title': 'Zureeal on TikTok',
+            'thumbnail': 'http://m-p16.akamaized.net/img/tos-maliva-p-0068/5e7a4ec40fb146888fa27aa8d78f86fd~noop.image',
+            'description': '#bowsette#mario#cosplay#uk#lgbt#gaming#asian#bowsettecosplay',
+            'uploader': 'Zureeal',
+            'width': 540,
+            'height': 960,
+        }
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        data = self._parse_json(
+            self._search_regex(
+                r'var\s*data\s*=\s*({.+?});', webpage, 'data'
+            ), video_id)
+
+        title = self._og_search_title(webpage)
+
+        description = str_or_none(try_get(data, lambda x: x['desc']))
+        width = int_or_none(try_get(data, lambda x: x['video']['width']))
+        height = int_or_none(try_get(data, lambda x: x['video']['height']))
+
+        formats = []
+
+        for key, label in (('play_addr_lowbr', 'Low'), ('play_addr', 'Normal'), ('download_addr', 'Download')):
+            for format in try_get(data, lambda x: x['video'][key]['url_list']):
+                format_url = url_or_none(format)
+                if not format_url:
+                    continue
+                formats.append({
+                    'url': url,
```

Comment:
Invalid URL, have you even tested?

Reply:
Yes I have tested it and it worked. That's why I didn't notice I forgot to update the variable name.
```
python -m youtube_dl https://m.tiktok.com/v/6606727368545406213.html
[TikTok] 6606727368545406213: Downloading webpage
[download] Destination: Zureeal on TikTok-6606727368545406213.mp4
[download] 100% of 68.16KiB in 00:01
```